src/site/topic19.rst

```diff
@@ -125,6 +125,26 @@ Factorial
 Observations
 ============
 
+* Notice how in the call ``recursiveSum(4)``, we would have made a call to ``recursiveSum(3)``
+* If I ever call this function again, but want to check ``recursiveSum(5)``, we would have to call ``recursiveSum(4)`` again
+* In other words, to know ``recursiveSum(x)``, we will be answering ``recursiveSum(x - 1)``, ``recursiveSum(x - 2)``, ... ``recursiveSum(1)``, and ``recursiveSum(0)``
+
+* You may have also noticed the relationship between the Stack ADT we discussed and the call stack
+
+* Additionally, anything you do with iteration you can do with recursion, and anything you do with recursion you can do with iteration
+* However, just because you *can* doesn't mean you *should*
+    * Based on the design of the computational systems we use, recursion creates additional overhead that slows things down
+        * Creating frames
+        * Pushing/popping from the stack
+    * In many programming languages, the compilers will even optimize certain types of recursive functions by translating them to an iterative version
```

Comment:
I'd probably also emphasize that the simplicity of the program will also matter. You can always eek out more performance later. If the problem you're solving is easier for you to solve recursively, then do that first. When you have a grasp of the problem and some tests backing you up, you can revisit and ask if the program is performant enough.

Reply:
Was gonna' hit this hard when discussion Fibonacci, however I really like the idea of starting that conversation here.

Reply:
You can reinforce it there too, recursive fibonacci is a pretty good example of when you'd actually want to revisit.

Reply:
This is actually brilliant. 

"Sometimes good enough is good enough" -> "If performance ends up being an issue, then deal with it later" -> "Here's recursive fib --- yeah, gonna' wanna fix this one; there are speedups and then there are speedups"